tests/autorestart_test/test_container_autorestart.py

```diff
@@ -0,0 +1,243 @@
+
+def verify_no_autorestart_with_non_critical_process(duthost, container_name, process_name):
+    """
+    @summary: Killing a non-critical process in a container to verify whether the container 
+              is still in the running state 
+    """
+    is_running = is_container_running(duthost, container_name)
+    if is_running == "false":
+        pytest_assert(False, "{} is not running".format(container_name))
+
+    kill_process_by_name(duthost, container_name, process_name)
+
+    logging.info("Checking whether the {} is still running...".format(container_name))
+    pytest_assert(wait_until(CONTAINER_STOP_THRESHOLD_SECS, 
+                      CONTAINER_CHECK_INTERVAL_SECS,
+                      check_container_status, duthost, container_name, False),
+                  "{} is stopped unexpectedly".format(container_name))
+    logging.info("{} is running".format(container_name))
+
+def test_containers_autorestart(duthost):
+    container_autorestart_info = get_autorestart_container_and_state(duthost)
+    for container_name in container_autorestart_info:
+        if container_name in ["restapi"]:
+            continue
+        
+        logging.info("Change {} auto-restart state to 'enabled'".format(container_name))
+        duthost.shell("config container feature autorestart {} enabled".format(container_name))
+        
+        verify_no_autorestart_with_non_critical_process(duthost, container_name, "rsyslogd")
+
+        if container_name in ["pmon", "radv"]:
+            logging.info("Restore {} auto-restart state to {}".format(container_name, container_autorestart_info[container_name]))
+            duthost.shell("config container feature autorestart {} {}".format(container_name, container_autorestart_info[container_name]))
+            continue
+        
+        critical_group_list, critical_process_list = get_critical_group_and_process_list(duthost, container_name)
+        for critical_process in critical_process_list:
+            verify_autorestart_with_critical_process(duthost, container_name, critical_process, "", "", True)
+            break
+        for critical_group in critical_group_list:
+            group_program_info = get_group_program_info(duthost, container_name, critical_group) 
+            for program_name in group_program_info:
+                verify_autorestart_with_critical_process(duthost, container_name, "", program_name, 
+                                                         group_program_info[program_name], False)
+                break
+
+        if container_name in ["swss", "database"]:
+            logging.info("Sleep 10 seconds after testing the {}...".format(container_name))
+            time.sleep(10)
+
+        logging.info("Restore {} auto-restart state to {}".format(container_name, container_autorestart_info[container_name]))
+        duthost.shell("config container feature autorestart {} {}".format(container_name, container_autorestart_info[container_name]))
```

Comment:
Only need to change state if the previous state was 'disabled'

Reply:
Will fix.